### PR TITLE
Changes to store Daughters of Kshort and Lambda in the miniAOD prunedGenParticle Collection

### PR DESCRIPTION
**Pull request by gmelachr**
Modify prunedGenParticles_cfi.py to store daughters of Kshort and Lambda in the prunedGenParticle collection in order to use the daughter's production vertices for Kshort and Lambda lifetime measurements. The changes have been tested locally.

### CODE REVIEW

**Comment by simonepigazzini** [comment]
please test

**Comment by mandrenguyen** [comment]
The nano summary table just above doesn't seem to show any size increase.
Is the change only in mini?
@gmelachr what output size change did you observe in your local tests?  With what workflow and how many events?

**Comment by gmelachr** [comment]
Let me clarify a bit: I only updated the PhysicsTools/PatAlgos/python/slimming/prunedGenParticles_cfi.py to change-correct the available information in the miniAODs. Using the nominal framework, I was seeing the same production vertex for Kshort  and for its daughters. The daughters were stored in the packedGenParticles collection with a wrong production vertex and the Kshort in the prunedGenParticle Collection. Now, the daughters are also stored in the prunedGenParticleCollection with the correct production vertex. 

Furthermore, I didn't produce nanoAODs, I produced and checked things only for miniAODs locally. 
I produced 655 events for B0->J/ψΚshort->μμππ and 422 events for Λb->J/ψΛ0 ->μμpπ. I checked the size of the miniAODs (with ls -lh) and in both cases (the ones produced with the nominal version and the ones produced with the modified one) the miniAODs had the same size. 

Let me know if this responds your question, 
Giorgos

**Comment by simonepigazzini** [comment]
I enabled the nano tests more as a cross-check than anything else. I did not expect a change in size at the NANO level, since the gen level is further pruned at the nano step. I think that overall the impact of the changes is negligible also for MINI

**Comment by mandrenguyen** [comment]
+reconstruction
Thanks for the explanations.
Changes are acceptable for reco, provided they are for XPOG :-)

**Comment by gmelachr** [comment]
Hi, 

according to the last message I understand that the changes have been accepted. So, how are we going to proceed now? Are we waiting for a further approval by someone? Furthermore, what is the procedure to backport these changes to 12_4_X?

**Comment by perrotta** [comment]
@gmelachr we still miss the review of @cms-sw/xpog-l2 (whch are the main responsible for this code)

File: PhysicsTools/PatAlgos/python/slimming/prunedGenParticles_cfi.py [hunk 1]
```diff
@@ -18,7 +18,8 @@
 	"drop  pdgId == 21 && pt < 5",                                     # remove pythia8 garbage
         "drop   status == 2 && abs(pdgId) == 21",                          # but remove again gluons in the inheritance chain
         "keep abs(pdgId) == 23 || abs(pdgId) == 24 || abs(pdgId) == 25 || abs(pdgId) == 6 || abs(pdgId) == 37 ",   # keep VIP(articles)s
-        "keep abs(pdgId) == 310 && abs(eta) < 2.5 && pt > 1 ",                                                     # keep K0
+        "keep++ abs(pdgId) == 310 || abs(pdgId) == 3122",                  # Keep Kshort, Lambda0 and their daughters
+        "keep+ abs(pdgId) == 310 || abs(pdgId) == 3122",                   # but keep first daughter, to allow lifetime determinations
```
[the remark this answers is not in the document]
Comment (gmelachr) [reply PhysicsTools/PatAlgos/python/slimming/prunedGenParticles_cfi.py:22]: Hmm, I want to store all Kshorts and Lambdas for B0->X(μμ)Kshort(ππ), Β+ ->Χ(μμ)Κ*+ -->μμKshortπ-->μμπππ and Λb->Χ(mumu)Λ0(pπ) in order to measure acceptance. So, a possible cut for eta or for pT would affect the acceptance, wouldn't it? 

What is the problem with the current syntax?

[the remark this answers is not in the document]
Comment (simonepigazzini) [reply PhysicsTools/PatAlgos/python/slimming/prunedGenParticles_cfi.py:22]: ok understood. There is no problem, I was just trying to get to the minimal set of particles to add.

**Comment by simonepigazzini** [comment]
+1

**Comment by perrotta** [comment]
+1

**Comment by gmelachr** [comment]
> @cms-sw/pdmv-l2
> 
> my understanding is that the change is needed in 12_4_X to patch the ongoing production. If you agree to produce the required samples with a dedicated central production we can proceed with a backport to 12_4_X.

The productions for the above-mentioned decays have not been requested yet with central productions. I was waiting for the changed to be integrated in CMSSW in order to avoid double productions and possible problems. 

The current MC requests for BPAG are running on CMSSW_12_4_11_path3, so please proceed with the backport in order to proceed with the central production too.

thanks

**Comment by sunilUIET** [comment]
@simonepigazzini we will produce the samples as requested by pwgs. It is fine to backport it.

**Comment by simonepigazzini** [comment]
@gmelachr, please proceed with the backport. You just need to apply the same changes starting from CMSSW_12_4_11_path3 instead of a recent IB. 

Thank you

**Comment by gmelachr** [comment]
> @gmelachr, please proceed with the backport. You just need to apply the same changes starting from CMSSW_12_4_11_path3 instead of a recent IB.
> 
> Thank you

And then I will create a pull request to CMSSW_12_4_X, right?

**Comment by simonepigazzini** [comment]
yep

**Pull request merged**